aws-distro-opentelemetry-node-autoinstrumentation/src/patches/aws/services/bedrock.ts

```diff
@@ -211,11 +211,173 @@ export class BedrockRuntimeServiceExtension implements ServiceExtension {
       spanAttributes[AwsSpanProcessingUtil.GEN_AI_REQUEST_MODEL] = modelId;
     }
 
+    if (request.commandInput?.body) {
+      const requestBody = JSON.parse(request.commandInput.body);
+      if (modelId.includes('amazon.titan')) {
+        if (requestBody.textGenerationConfig?.temperature !== undefined) {
```

Comment:
Updated all conditionals to strictly check for `undefined`. 

The previous conditional pattern `if (requestBody.textGeneration?.temperature)` relies on truthy/falsy values which does not work as intended on edge cases where values are set to `0`.